src/openms/include/OpenMS/MATH/MISC/MathFunctions.h

```diff
@@ -333,7 +333,7 @@ namespace OpenMS
       if (q > 1.0) q = 1.;
 
       const auto n  = x.size();
-      const auto id = (n - 1) * q;
+      const auto id = std::max(0., n * q - 1); // -1 for c++ index starting at 0
```

Comment:
this could be dangerous:
what if I pass an empty vector and `q=0`?
since n and q are unsigned then `n*q - 1` is std::numeric_limits<size_t>::max()-1 ... and we will get a segfault.
Can we:
1) throw an exception if `x` is empty (required for any value of q actually)
2) make sure `q` is a double by removing the template parameter.